mobilecoind/src/transaction_memo.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) 2018-2024 The MobileCoin Foundation
+
+use mc_account_keys::AccountKey;
+use mc_api::ConversionError;
+use mc_mobilecoind_api::{
+    mobilecoind_api, TransactionMemo_RTH_oneof_payment_id, TransactionMemo_oneof_transaction_memo,
+};
+use mc_transaction_builder::{
+    BurnRedemptionMemoBuilder, EmptyMemoBuilder, MemoBuilder, RTHMemoBuilder,
+};
+use mc_transaction_extra::{BurnRedemptionMemo, SenderMemoCredential};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TransactionMemo {
+    /// Recoverable Transaction History memo with an optional u64 specifying the
+    /// subaddress index to generate the sender memo credential from
+    RTH {
+        /// Optional subaddress index to generate the sender memo credential
+        /// from.
+        subaddress_index: Option<u64>,
+    },
+
+    RTHWithPaymentIntentId {
```

Comment:
```suggestion
    RthWithPaymentIntentId {
```

No clippy error, just carrying through the naming convention...